make error

DKMS make.log for cryptodev-1.10 for kernel 4.4.176-1-ARCH (armv5tel)
Fri Aug 16 07:40:52 UTC 2019
/var/lib/dkms/cryptodev/1.10/build/zc.c: In function '__get_userbuf':
/var/lib/dkms/cryptodev/1.10/build/zc.c:64:44: warning: passing argument 7 of 'get_user_pages' from incompatible pointer type [-Wincompatible-pointer-types]
    (unsigned long)addr, pgcount, write, 0, pg, NULL);
                                            ^~
In file included from /var/lib/dkms/cryptodev/1.10/build/zc.c:28:
include/linux/mm.h:1225:31: note: expected 'struct vm_area_struct **' but argument is of type 'struct page **'
       struct vm_area_struct **vmas);
       ~~~~~~~~~~~~~~~~~~~~~~~~^~~~
/var/lib/dkms/cryptodev/1.10/build/zc.c:63:8: error: too many arguments to function 'get_user_pages'
  ret = get_user_pages(task, mm,
        ^~~~~~~~~~~~~~
In file included from /var/lib/dkms/cryptodev/1.10/build/zc.c:28:
include/linux/mm.h:1222:6: note: declared here
 long get_user_pages(struct task_struct *tsk, struct mm_struct *mm,
      ^~~~~~~~~~~~~~
make[2]: *** [scripts/Makefile.build:278: /var/lib/dkms/cryptodev/1.10/build/zc.o] Error 1
make[1]: *** [Makefile:1436: _module_/var/lib/dkms/cryptodev/1.10/build] Error 2
make: *** [Makefile:27: build] Error 2

--- a/zc.c
+++ b/zc.c
@@ -63,9 +63,12 @@ int __get_userbuf(uint8_t __user *addr, uint32_t len, int write,
 #else
 	mmap_read_lock(mm);
 #endif
-#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0))
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(4, 4, 168))
 	ret = get_user_pages(task, mm,
 			(unsigned long)addr, pgcount, write, 0, pg, NULL);
+#elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 6, 0))
+	ret = get_user_pages(task, mm,
+			(unsigned long)addr, pgcount, write, pg, NULL);
 #elif (LINUX_VERSION_CODE < KERNEL_VERSION(4, 9, 0))
 	ret = get_user_pages_remote(task, mm,
 			(unsigned long)addr, pgcount, write, 0, pg, NULL);

